test/dependency-manager/interpolation.test.ts

```diff
@@ -1315,6 +1315,60 @@ describe('interpolation spec v1', () => {
     });
   });
 
+  it('interpolate interfaces ingress whitelist', async () => {
+    const component_config = `
+    name: examples/hello-world
+    parameters:
+      ip_whitelist:
+        default: [127.0.0.1]
+      required_ip_whitelist:
+    interfaces:
+      api:
+        url: \${{ services.api.interfaces.main.url }}
+        ingress:
+          whitelist: \${{ parameters.ip_whitelist }}
+      api2:
+        url: \${{ services.api.interfaces.main.url }}
+        ingress:
+          whitelist: \${{ parameters.required_ip_whitelist }}
+    services:
+      api:
+        interfaces:
+          main: 8080
+    `
+
+    mock_fs({
+      '/stack/architect.yml': component_config,
+    });
+
+    const manager = new LocalDependencyManager(axios.create(), {
+      'examples/hello-world': '/stack/architect.yml',
+    });
+    const config = await manager.loadComponentConfig('examples/hello-world');
+    const graph = await manager.getGraph(
+      await manager.loadComponentConfigs(config),
+      // @ts-ignore
```

Comment:
Remove ts-ignore and make values.yml support more than just strings? Curious for your thoughts here.

Reply:
I hadn't considered this case. Do you see any reasons not to enable something like this besides the fact that this might be changing very soon? I mostly imagined this feature being hardcoded though because it's not something that someone would use for local development, and I don't really expect it to change frequently, if at all, for remote deployments